unity/src/test/scala/io/delta/unity/InMemoryUCClientSuite.scala

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.unity
+
+import java.lang.{Long => JLong}
+import java.net.URI
+import java.util.Optional
+
+import scala.collection.JavaConverters._
+
+import io.delta.storage.commit.CommitFailedException
+import io.delta.storage.commit.uccommitcoordinator.InvalidTargetTableException
+
+import org.scalatest.funsuite.AnyFunSuite
+
+/** Unit tests for [[InMemoryUCClient]]. */
+class InMemoryUCClientSuite extends AnyFunSuite with UCCatalogManagedTestUtils {
+
+  private def testGetCommitsFiltering(
+      allVersions: Seq[Long],
+      startVersionOpt: Optional[JLong],
+      endVersionOpt: Optional[JLong],
+      expectedVersions: Seq[Long]): Unit = {
+    val client = getInMemoryUCClientWithCommitsForTableId("tableId", allVersions)
+    val response = client.getCommits("tableId", fakeURI, startVersionOpt, endVersionOpt)
+    val actualVersions = response.getCommits.asScala.map(_.getVersion)
+
+    assert(actualVersions == expectedVersions)
+  }
+
+  test("TableData::appendCommit throws if commit version is not maxRatifiedVersion + 1") {
+    val tableData = new InMemoryUCClient.TableData
+    val commit = createCommit(1L)
+
+    val exMsg = intercept[CommitFailedException] {
+      tableData.appendCommit(commit)
+    }.getMessage
+
+    assert(exMsg.contains("Expected commit version 0 but got 1"))
+  }
+
+  test("TableData::appendCommit appends the commit and updates the maxRatifiedVersion") {
+    val tableData = new InMemoryUCClient.TableData
+    tableData.appendCommit(createCommit(0L))
+
+    assert(tableData.getMaxRatifiedVersion == 0L)
+    assert(tableData.getCommits.size == 1)
+    assert(tableData.getCommits.head.getVersion == 0L)
+
+    tableData.appendCommit(createCommit(1L))
+    assert(tableData.getMaxRatifiedVersion == 1L)
+    assert(tableData.getCommits.size == 2)
+    assert(tableData.getCommits.last.getVersion == 1L)
+  }
+
+  test("getCommits throws InvalidTargetTableException for non-existent table") {
+    val client = new InMemoryUCClient("ucMetastoreId")
+    val tableId = "non-existent-table"
+
+    val exception = intercept[InvalidTargetTableException] {
+      client.getCommits(tableId, new URI("s3://bucket/table"), Optional.empty(), Optional.empty())
+    }
+    assert(exception.getMessage.contains(s"Table not found: $tableId"))
```

Comment:
```suggestion
    assert(exception.getMessage.contains(s"Table not found: non-existent-table"))
```

nit: sorry if i missed this, but try not to replicate business logic in tests.

Reply:
Hey! Could you actually explain this a bit more to me? I'd like to understand this best practice more

i.e. how this was replicating business logic? and why that was bad or why printing out `non-existent-table` is better?

Reply:
Its replicating the substitution of the table name to construct the error message.  printing out non-existent-table means no additional steps are needed to verify what the output looks like.  So I can say yes the error message looks good or not.

One could argue that you want to show the table ID shows up in the error message.  This could be handled by splitting up the assert into two parts (either the full message hard-coded + an assert the message contains `tableId` or just two asserts that the message contains both the prefix and suffix).

https://testing.googleblog.com/2014/07/testing-on-toilet-dont-put-logic-in.html is I think where I originally got this idea from.  Of course there are no hard and fast rules.